editor/components/meta-boxes/meta-boxes-area/index.js

```diff
@@ -84,30 +74,13 @@ class MetaBoxesArea extends Component {
 		this.node.appendChild( this.form );
 		this.form.onSubmit = ( event ) => event.preventDefault();
 		this.originalFormData = this.getFormData();
-		this.form.addEventListener( 'change', this.checkState );
-		this.form.addEventListener( 'input', this.checkState );
 		this.props.metaBoxLoaded( location );
 	}
 
 	getFormData() {
 		return jQuery( this.form ).serialize();
 	}
 
-	checkState() {
-		const { loading } = this.state;
-		const { isDirty, changedMetaBoxState, location } = this.props;
-
-		const newIsDirty = ! isEqual( this.originalFormData, this.getFormData() );
```

Comment:
Seems `this.originalFormData` is defunct now and can be removed, along with `getFormData`.